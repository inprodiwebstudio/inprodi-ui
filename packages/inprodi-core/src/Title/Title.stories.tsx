import React                             from "react";
import { ComponentMeta, ComponentStory } from "@storybook/react";
import { Title, TitleProps }             from "./Title";

export default {
	title     : "Core/Typography/Title",
	component : Title,
	argTypes  : {
		color : { control : "color" },
	},
} as ComponentMeta<typeof Title>;

const Template: ComponentStory<typeof Title> = (args : TitleProps) => {

	return(
		<Title {...args} >
			Hola
		</Title>
	);
};
// Title font-size: h1-h6 or any valid CSS font-size value
// MantineNumberSize
export const H1 = Template.bind({});
H1.args = {
	order : 1,
};
export const H2 = Template.bind({});
H2.args = {
	order : 2,
};
export const H3 = Template.bind({});
H3.args = {
	order : 3,
};
export const H4 = Template.bind({});
H4.args = {
	order : 4,
};
export const H5 = Template.bind({});
H5.args = {
	order : 5,
};
export const H6 = Template.bind({});
H6.args = {
	order : 6,
};
