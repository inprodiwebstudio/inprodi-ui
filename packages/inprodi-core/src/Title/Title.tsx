import {
	TitleOrder,
	MantineNumberSize,
	Title as MantineTitle,
	TitleProps as MantineTitleProps,
} from "@mantine/core";

export interface TitleProps extends MantineTitleProps {
    size ?: MantineNumberSize;
}
export const Title = ({children, size = "md", color, ...rest} : TitleProps) => {
	const getOrderTitle = () => {
		const orders = {
			"xl" : 2,
			"lg" : 3,
			"md" : 4,
			"sm" : 5,
			"xs" : 6,
		} as { [key in MantineNumberSize] : TitleOrder };
		return orders[size];
	};
	return (
		<MantineTitle
			order={getOrderTitle()}
			sx={(theme) => ({
				color : color ?
					color : theme.colorScheme === "dark" ? "#fff" : "#000",
			})}
			{...rest}
		>
			{children}
		</MantineTitle>
	);
};

export default Title;
