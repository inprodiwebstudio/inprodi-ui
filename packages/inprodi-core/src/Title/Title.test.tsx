import React      from "react";
import { render } from "@testing-library/react";
import { Title }  from "./Title";

describe("Title", () => {
	test("should snapshot", () => {
		const { container } = render(
			<Title>My title</Title>
		);
		expect(container).toMatchSnapshot();
	});
	test("should h1 order", () => {
		const { getByRole } = render(
			<Title order={1}>My title</Title>
		);
		const h1 = getByRole("heading");
		expect(h1.tagName).toBe("H1");
	});
	test("should h2 order", () => {
		const { getByRole } = render(
			<Title size="xl">My title</Title>
		);
		const h2 = getByRole("heading");
		expect(h2.tagName).toBe("H2");
	});
	test("should h3 order", () => {
		const { getByRole } = render(
			<Title size="lg">My title</Title>
		);
		const h3 = getByRole("heading");
		expect(h3.tagName).toBe("H3");
	});
	test("should h4 order", () => {
		const { getByRole } = render(
			<Title size="md">My title</Title>
		);
		const h4 = getByRole("heading");
		expect(h4.tagName).toBe("H4");
	});
	test("should h5 order", () => {
		const { getByRole } = render(
			<Title size="sm">My title</Title>
		);
		const h5 = getByRole("heading");
		expect(h5.tagName).toBe("H5");
	});
	test("should h6 order", () => {
		const { getByRole } = render(
			<Title size="xs">My title</Title>
		);
		const h6 = getByRole("heading");
		expect(h6.tagName).toBe("H6");
	});
	test("should has custom className", () => {
		const className = "custom-className";
		const { getByRole } = render(
			<Title className={className}>My title</Title>
		);
		const title = getByRole("heading");
		expect(title.className.includes(className)).toBeTruthy();
	});
});
