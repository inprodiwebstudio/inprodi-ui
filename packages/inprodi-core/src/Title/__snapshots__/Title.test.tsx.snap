// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title should snapshot 1`] = `
<div>
  <h4
    class="mantine-Text-root mantine-Title-root mantine-cmra41"
  >
    My title
  </h4>
</div>
`;
